src/Http/Routing/src/Matching/DfaMatcherBuilder.cs

```diff
@@ -670,6 +698,10 @@ private void ApplyPolicies(DfaNode node)
                 return;
             }
 
+            // We're done with the precedence based work. Sort the endpoints
+            // before applying policies for simplicity in policy-related code.
+            node.Matches.Sort(_comparer);
```

Comment:
Is this sort verified by any tests?

The tests you've added don't have policies. Do existing policy tests fail if this is removed?

Reply:
It wouldn't hurt to add a test for it, but I expect something would have failed. We don't  re-sort these in our policy implementations.

Reply:
I'd like to test please 😄

We're potentially passing the matches to a user implemented policy. I think we should verify the order so we don't accidentally change it now or in the future.

Reply:
I've added a new test oriented specifically at the sorting issue.